sink/datadog.go

```diff
@@ -470,6 +494,35 @@ func (s *Datadog) estimateSize(metrics []datadogV2.MetricSeries) (int, error) {
 	return size / len(metrics), nil
 }
 
+func (s *Datadog) uniqIdentifierForMetric(name string, groups map[string]string) string {
+	sortedGroupValues := s.valuesInSortedKeyOrder(groups)
+
+	var key string
+	key += name
+	key += sortedGroupValues
+
+	return key
+}
+
+func (s *Datadog) valuesInSortedKeyOrder(tags map[string]string) string {
```

Comment:
Don't split out functions when they're not used elsewhere. This func _is_  the bulk of `uniqIdentifierForMetric` (or `metricId` as I'd call it). This kind separation immediately raises the question (to future devs): "Is  valuesInSortedKeyOrder used elsewhere? or other things?" But no, it's not.

Reply:
Inlined 👍